README.md

```diff
@@ -541,3 +816,292 @@ Semantic release will only release on configured branches, so it is safe to run
 ## License
 
 MIT © [Sanity.io](https://www.sanity.io/)
+
+# Migrate
+
+## From `v4`
+
+### No longer shipping `ES5`<!-- omit in toc -->
+
+The target is changed to [modern browsers] that supports `ES6` `class`, `{...rest}` syntax and more. You may need to update your bundler to a recent major version. Or you could configure your bundler to transpile `@sanity/client`, and `get-it`, which is the engine that powers `@sanity/client` and uses the same output target.
+
+### Node.js `v12` no longer supported<!-- omit in toc -->
+
+Upgrade to the [LTS release, or one of the Maintenance releases](https://github.com/nodejs/release#release-schedule).
+
+### The `default` export is replaced with the named export `createClient`<!-- omit in toc -->
```

Comment:
This is the big one. I think it should be called out explicitly in the release notes, and not just link here. It will hit _everyone_ hard.

Reply:
This is the big one indeed. What is a good way to go about this? Add something like this? 

## Breaking change: 4.x

We have removed the default export and replaced it with a named one: 

```js
// Before Sanity Client v4
import SanityClient from '@sanity/client'

// After Sanity Client v4
import {createConfig} from '@sanity/client'
```

This also requires us to update all documentation and guides that have the client import.

Reply:
I've udpated the PR description and the commit body that'll be used. Semantic release will render something like this as release notes from it:

## [5.0.0](https://github.com/sanity-io/client/compare/v4.0.1...v5.0.0) (2023-01-13)


### ⚠ BREAKING CHANGES

We have removed the default export and replaced it with a named one:
  ```diff
  -import SanityClient from '@sanity/client'
  +import {createConfig} from '@sanity/client'
  ```
[The migration guide outlines every breaking change and how to migrate your code](https://github.com/sanity-io/client/tree/modernize?rgh-link-date=2023-01-09T16%3A08%3A01Z#from-v4)

### Features

- full ESM support in modern runtimes and tooling (Bun, Deno, Edge, without breaking Node.js ESM and CJS compatibility)
- codebase rewritten in TypeScript, typings are generated and no longer manually maintained
- make `httpRequest` on `SanityClient` configurable, intended for libraries wishing to extend the client
- shipping modern syntax, reducing bundlesize

Reply:
We can enrich the release notes and changelog entry after semantic release have published it. But if we add more to the commit body we'll end up with Semantic Release arbitrarily cropping/mangling the output.

Reply:
Btw @kmelve  the new version will be `v5`. [`v4` is already released](https://github.com/sanity-io/client/releases/tag/v4.0.0), and don't have a migration guide as it didn't contain any API changes, it only added `pkg.exports` to add partial ESM support for runtimes that require it (like Cloudflare Workers and Edge Runtime on Next.js)

Reply:
>This is the big one indeed. What is a good way to go about this? 

Perhaps we could create a companion codemod for the imports change: https://github.com/facebook/jscodeshift @rexxars what do you think?

Reply:
We could do that. Do you want me to write one?